ipaserver/install/ipa_migrate.py

```diff
@@ -622,7 +622,7 @@ def write_update_to_ldif(self, entry, add_entry=False):
                 else:
                     action = "replace"
                 ldif_entry += f"{action}: {attr}\n"
-                for val in vals:
+                for val in [val for val in (vals or [])]:
```

Comment:
Lint reports the following: 

```
************* Module ipaserver.install.ipa_migrate
ipaserver/install/ipa_migrate.py:625: [R1721(unnecessary-comprehension), IPAMigrate.write_update_to_ldif] Unnecessary use of a comprehension, use list(vals or []) instead.)


```

Reply:
or: `if not vals: continue` way higher otherwise you could end up with a bare delete or replace in the ldif with no value right?

Reply:
> or: `if not vals: continue` way higher otherwise you could end up with a bare delete or replace in the ldif with no value right?

We need the line (624) above to write to ldif_entry, otherwise it would not record delete operations.

Reply:
Fair point about delete, I missed that, but what happens if a replace comes in with no val? Is an empty replace treated as a delete in ldif?

Reply:
> Fair point about delete, I missed that, but what happens if a replace comes in with no val? Is an empty replace treated as a delete in ldif?

I don't think the ipa-migrate tool would create a replace op with no value.  It only "replaces" if there a difference.  If there is no value it takes a different route.